### ISSUE TITLE
states "do not process large programs all at once"

### ISSUE THREAD

**Issue by Hardcore-fs** [yegord/snowman#200]
fine....

but WHY is the  default set to "decompile automatically"

**Comment by yegord** [comment]
Because this is what is most useful for the IDA plugin, and the plugin and the standalone decompiler share the code.

You can change the settings to whatever you prefer, close the window, and it will be remembered.

**Comment by Hardcore-fs** [comment]
nope.... it continually resets back to  "automatically" for the standalone program

**Comment by yegord** [comment]
What OS do you run it on?

**Comment by prpercival** [comment]
Sorry for resurrecting, but I also have this issue on Windows 10 Build 1909. It runs automatically and then crashes. If I set change from "Decompile Automatically" to "Decompile" it crashes and does not remember that selection on re-running.

**Comment by prpercival** [comment]
This is the last log before crash.
![image](https://user-images.githubusercontent.com/48492835/94748324-a671c500-0346-11eb-9422-05b4be7179f8.png)

**Comment by yegord** [comment]
You can start decompiler, uncheck "Decompile Automatically", close it, — and your selection will be saved.
Also, you can press Ctrl-Z to stop decompilation that is already running.

**Comment by prpercival** [comment]
Here's a gif of what I am experiencing. Same thing happens if I click Decompile. I did just realize that I can unselect "Decompile Automatically" before opening my file. If I do that then I am able to select "Decompile" after opening. I guess that is where I was going wrong?
![SnowmanBug](https://user-images.githubusercontent.com/48492835/94749511-94455600-0349-11eb-8039-3dd8f8d66754.gif)

**Comment by yegord** [comment]
Decompiling large programs as a whole is usually a bad idea: the decompiler goes out of memory and dies.
Consider decompiling smaller blocks of code (e.g., functions): select a range of instructions, push Ctrl-E.

**Comment by Hardcore-fs** [comment]
> What OS do you run it on?
osx
